src/com/intellij/idea/plugin/hybris/actions/AbstractCopyFileToHybrisConsoleAction.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * This file is part of "SAP Commerce Developers Toolset" plugin for Intellij IDEA.
+ * Copyright (C) 2019 EPAM Systems <hybrisideaplugin@epam.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.intellij.idea.plugin.hybris.actions;
+
+import com.intellij.execution.console.ConsoleExecutionEditor;
+import com.intellij.execution.console.LanguageConsoleImpl;
+import com.intellij.ide.projectView.ProjectView;
+import com.intellij.ide.projectView.impl.AbstractProjectViewPane;
+import com.intellij.idea.plugin.hybris.tools.remote.console.HybrisConsole;
+import com.intellij.idea.plugin.hybris.tools.remote.console.HybrisConsoleToolWindowFactory;
+import com.intellij.idea.plugin.hybris.tools.remote.console.view.HybrisConsolePanel;
+import com.intellij.idea.plugin.hybris.tools.remote.console.view.HybrisConsolePanelView;
+import com.intellij.openapi.actionSystem.AnAction;
+import com.intellij.openapi.actionSystem.CommonDataKeys;
+import com.intellij.openapi.actionSystem.DataContext;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.wm.ToolWindow;
+import com.intellij.openapi.wm.ToolWindowManager;
+import org.jetbrains.annotations.NotNull;
+
+import javax.swing.tree.TreePath;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public abstract class AbstractCopyFileToHybrisConsoleAction extends AnAction {
+
+    public boolean isRequiredFileExtension(
+        @NotNull final DataContext dataContext,
+        final String fileExtension,
+        final boolean oneFile
+    ) {
+        boolean isImpex = false;
+        final List<String> filesName = getFilesName(dataContext);
+        if (oneFile) {
+            return filesName.size() == 1 && filesName.get(0).contains(fileExtension);
+        } else {
+            for (String name : filesName) {
+                if (name.contains(fileExtension)) {
```

Comment:
We need to have an access to VirtualFile and then check its extension.
Smth like
```
StructureTreeModel.Node lastPathComponent = (StructureTreeModel.Node) file.getLastPathComponent();
ProjectViewNode element = (ProjectViewNode) lastPathComponent.getElement();
element.getVirtualFile().getExtension();
```